packages/by-name/ociLayerTar/package.nix

```diff
@@ -49,8 +49,11 @@ runCommandLocal "ociLayer"
 
     # Copy files into the tree (./root/)
     for i in ''${!srcs[@]}; do
+        # resolve symlinks
+        src=$(readlink -f ''${srcs[i]})
```

Comment:
I can't find any reasoning about this in the PR description. I agree with @burgerdev comment. What symlink isn't working for this fix to be needed?

Reply:
Commit description, sorry. It's about kata-kernel, which in the NixOS toplevel is a symlink to the bzImage in a separate derivation for the kernel.

Reply:
Sorry, I don't understand: the "derivation for the kernel" should also be installed in the system, shouldn't it, making the symlink resolve just fine?

Reply:
This is not about stuff in the system. We pack this into an OCI image layer, which is why I think adding support for resolving symlinks would be too complex compared to what we actually use this for. We select the kernel file from the "top-level" derivation to be included in the OCI layer. But it's just a symlink to a file in the derivation of the kernel. The OCI builder we have just copies the file without checking what it is nor dereferencing it. Thus, this - as of now - is an easy-to-trigger footgun for people using this. As we specify the target location in the layer for each file anyway, I think the proposed solution is just the best we can get with low complexity. I'm very strongly against keeping it as-is, just because the status quo is not sound imo.

Reply:
I don't like this (very sharp) edge of `ociLayerTar`, but I'll just drop the symlinks then..

Reply:
Yes, I understand - it's entirely possible that the idea of `ociLayerTar` is flawed when combined with nix. We could consider adding the nix store paths to the image and intentionally add only symlinks to the /opt/edgeless tree. But I guess we can figure that out some other time.